### DISCUSSION TITLE
为什么一个script只能有一个fun

### DISCUSSION THREAD

**Discussion by livelybug** [NonceGeek/Move-Camp#159] [Q&A] [answered]
或者这样设计的考虑是什么？

**Comment by guijunchen** [comment]
一个 script 块必须在开头声明[use](https://move-dao.github.io/move-book-zh/uses.html)，然后是[constants](https://move-dao.github.io/move-book-zh/constants.html)的内容,最后声明主函数 [function](https://move-dao.github.io/move-book-zh/functions.html)。主函数的名称可以是任意的(也就是说，它不一定命名为 main)，是script block中唯一的函数，可以有任意数量的参数，并且不能有返回值

**Comment by livelybug** [comment]
谢谢。我把问题扩展一下: 假如MOVE标准规定一个 script 块里可以包含两个或两个以上的function， 这样的规定会有什么问题吗？

**Reply by guijunchen** [reply]
我理解几个原因： 1、script只能调用module函数 2、script不能自己实现改变全局状态的功能，只能通过调用module函数。3、aptos 上script可以当作参数传给module 函数调用。  特别是第二点来说就没有必要存在第二个函数

**Comment by livelybug** [comment, marked as answer]
Reply 好像不能作为答案 ：）
只好这样： 答案是 https://github.com/NonceGeek/Web3-dApp-Camp/discussions/159#discussioncomment-4024127
